### PR TITLE
Drop cache in executors in addition to the driver

### PR DESCRIPTION
**Pull request by zhangweilst**
## Changes

Drop the system cache every time before a query starts, both in driver and executors.

## Test
Manually tested.

### CODE REVIEW

**Comment by zhangweilst** [comment]
@yueawang This pr drops the cache both in executors and driver, and each time before a query starts. The tests show that queries that disabled the oap cache benefits a lot from system cache. Ready to have your opinion!

File: src/main/scala/org/apache/spark/sql/OapTestSuite.scala [hunk 1]
```diff
@@ -113,6 +116,13 @@ abstract class OapTestSuite extends BenchmarkConfigSelector with OapPerfSuiteCon
   def resultMap = _resultMap
 
   protected def testSet: Seq[OapBenchmarkTest]
+  protected def dropCache(): Unit = {
+    val nodes = spark.sparkContext.getExecutorMemoryStatus.map(_._1.split(":")(0))
+    nodes.foreach { node =>
+      val dropCacheResult = Seq("bash", "-c", s"""ssh $node "echo 3 > /proc/sys/vm/drop_caches"""").!
```
Comment (yueawang) [inline src/main/scala/org/apache/spark/sql/OapTestSuite.scala]: Does it have privilege issue?? Can sudo work remotely??

File: src/main/scala/org/apache/spark/sql/OapTestSuite.scala [hunk 2]
```diff
@@ -113,6 +116,13 @@ abstract class OapTestSuite extends BenchmarkConfigSelector with OapPerfSuiteCon
   def resultMap = _resultMap
 
   protected def testSet: Seq[OapBenchmarkTest]
+  protected def dropCache(): Unit = {
+    val nodes = spark.sparkContext.getExecutorMemoryStatus.map(_._1.split(":")(0))
```
Comment (yueawang) [inline src/main/scala/org/apache/spark/sql/OapTestSuite.scala]: This looks trivial, do we have an elegant way to do that?

**Review by yueawang — COMMENTED** [review]
Thanks @zhangweilst , very good catch!
Just few comments.

File: src/main/scala/org/apache/spark/sql/OapTestSuite.scala (hunk 1 above)
Comment (zhangweilst) [reply src/main/scala/org/apache/spark/sql/OapTestSuite.scala]: We run the script in driver with sudo or root (effectively runs as super user), and ssh sure can work remotely.

File: src/main/scala/org/apache/spark/sql/OapTestSuite.scala (hunk 2 above)
Comment (zhangweilst) [reply src/main/scala/org/apache/spark/sql/OapTestSuite.scala]: Yeah...kind of a ugly hack. I didn't come up with another idea. Will try to find out if there's a better way.

**Pull request merged**